_ext/events.rb

```diff
@@ -0,0 +1,51 @@
+require 'aweplug/helpers/searchisko'
+require 'parallel'
+
+module JBoss
+  module Developer
+    class Events
+
+      # def initialize
+      #
+      # end
+
+      def execute site
+        searchisko = Aweplug::Helpers::Searchisko.default site, 0
+
+        Parallel.each(site.events, in_threads: (site.build_threads || 0)) do |(id, data)|
+
+        searchisko_hash = data.collect do |(key, value)|
+
+          case key
+            when 'title'
+              [:sys_title, value]
+            when 'description'
+              [:sys_description, value]
+            when 'on_homepage'
+              [:featured, value]
+            # Hacked together
+            when 'type'
+              [:sys_url_view, value]
+            when 'more_details'
+              [:sys_url_view, value]
+            else
+              [key, value]
+          end
+        end.to_h
+
+        id_gen = searchisko_hash[:sys_title].to_s.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].to_s.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')
+
+        # Add identifier and other required fields for Searchisko Content Object
+        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + id_gen.to_s})
+
+        unless !site.push_to_searchisko || site.profile =~ /development/
+          searchisko.push_content('jbossdeveloper_events',
```

Comment:
Should this be 'jbossdeveloper_event' (no plural). Notice what is used elsewhere.